src/util.ts

```diff
@@ -350,16 +350,16 @@ function resolveToolsGopath(): string {
 
 	let toolsGopathForWorkspace = vscode.workspace.getConfiguration('go')['toolsGopath'] || '';
 
-	// In case of single root, use resolvePath to resolve ~ and ${workspaceRoot}
+	// In case of single root
 	if (!vscode.workspace.workspaceFolders || vscode.workspace.workspaceFolders.length <= 1) {
 		return resolvePath(toolsGopathForWorkspace);
 	}
 
-	// In case of multi-root, resolve ~ and ignore ${workspaceRoot}
+	// In case of multi-root, resolve ~ and ${workspaceFolder}
 	if (toolsGopathForWorkspace.startsWith('~')) {
 		toolsGopathForWorkspace = path.join(os.homedir(), toolsGopathForWorkspace.substr(1));
 	}
-	if (toolsGopathForWorkspace && toolsGopathForWorkspace.trim() && !/\${workspaceRoot}/.test(toolsGopathForWorkspace)) {
+	if (toolsGopathForWorkspace && toolsGopathForWorkspace.trim() && !/\${workspaceFolder}/.test(toolsGopathForWorkspace)) {
```

Comment:
we should support both `$workspaceRoot` and `$workspaceFolder` here. Not everyone would have updated their settings.